t/43-scheduling-and-worker-scalability.t

```diff
@@ -0,0 +1,194 @@
+# is assigned to constant)
+BEGIN {
+    $ENV{OPENQA_SCHEDULER_MAX_JOB_ALLOCATION} = $ENV{SCALABILITY_FULLSTACK_JOB_COUNT}
+      // $ENV{SCALABILITY_FULLSTACK_WORKER_COUNT};
+}
+
+# read number of workers to spawn from environment variable; skip test entirely if variable not present
+# similar to other fullstack tests
+my $worker_count = $ENV{SCALABILITY_FULLSTACK_WORKER_COUNT};
+my $job_count    = $ENV{SCALABILITY_FULLSTACK_JOB_COUNT} // $worker_count;
+plan skip_all => 'set e.g. SCALABILITY_FULLSTACK_WORKER_COUNT=n to run the scalability test for n workers'
+  unless looks_like_number($worker_count) && $worker_count > 0;
+
+# setup basedir, config dir and database
+my $tempdir = setup_fullstack_temp_dir('scalability');
+my $schema  = OpenQA::Test::Database->new->create(skip_fixtures => 1);
+my $workers = $schema->resultset('Workers');
+my $jobs    = $schema->resultset('Jobs');
+
+# create web UI and websocket server
+my $mojoport              = Mojo::IOLoop::Server->generate_port();
+my $web_socket_server_pid = create_websocket_server($mojoport + 1, 0, 1, 1);
+my $webui_pid             = create_webapi($mojoport, sub { });
+
+# prepare spawning workers
+my $sharedir        = setup_share_dir($ENV{OPENQA_BASEDIR});
+my $resultdir       = path($ENV{OPENQA_BASEDIR}, 'openqa', 'testresults')->make_path;
+my $api_credentials = create_user_for_workers;
+my $api_key         = $api_credentials->key;
+my $api_secret      = $api_credentials->secret;
+my $webui_host      = "http://localhost:$mojoport";
+my $worker_path     = path($FindBin::Bin)->child('../script/worker');
+my $isotovideo_path = path($FindBin::Bin)->child('dummy-isotovideo.sh');
+my $worker_flags    = '--verbose --no-cleanup';
+my $worker_args
+  = "--apikey=$api_key --apisecret=$api_secret --host=$webui_host --isotovideo=$isotovideo_path $worker_flags";
+note("share dir: $sharedir");
+note("resultdir: $resultdir");
+
+# spawn workers
+note("Spawning $worker_count workers");
+sub spawn_worker {
+    my ($instance) = @_;
+
+    note("Starting worker '$instance'");
+    my $workerpid = fork();
+    return $workerpid if $workerpid != 0;
+
+    exec("perl $worker_path --instance=$instance $worker_args");
+    die "failed to start worker $instance";
+}
+my %worker_ids;
+my @worker_pids;
+push @worker_pids, spawn_worker($_) for 1 .. $worker_count;
+
+# create jobs
+note("Creating $job_count jobs");
+sub log_jobs {
+    my @job_info
+      = map { sprintf("id: %s, state: %s, result: %s, reason: %s", $_->id, $_->state, $_->result, $_->reason // 'none') }
+      $jobs->search({}, {order_by => 'id'});
+    note("All jobs:\n - " . join("\n - ", @job_info));
+}
+my %job_ids;
+my @job_settings = (
+    BUILD   => '0048@0815',
+    DISTRI  => 'opensuse',
+    VERSION => 'Factory',
+    FLAVOR  => 'tape',
+    ARCH    => 'x86_64',
+    MACHINE => 'xxx',
+);
+$job_ids{$jobs->create({@job_settings, TEST => "dummy-$_"})->id} = 1 for 1 .. $job_count;
+
+# wait one second per worker/job
```

Comment:
isn't it 0.1s? I would not replicate the value in the comment but describe the intention or reason

Reply:
0.1 is the poll interval. We try polling at most 10 times per worker/job which means we're waiting at most 1 second per worker/job. I'll turn that comment into `my $seconds_to_wait_per_worker_or_job = 1.0;` and use that variable in the computations below.

Reply:
great!